can/interfaces/usb_can_analyzer/usb_can_analyzer.py

```diff
@@ -0,0 +1,276 @@
+
+        byte_msg.append(0x12)     # Initialization Message ID
+
+        byte_msg.append(CanAnalyzer.BITRATE[self.bit_rate])  # CAN Baud Rate
+        byte_msg.append(CanAnalyzer.FRAMETYPE[self.frame_type]) 
+
+        for i in range(0, 4):
+            byte_msg.append(self.filter_id[i])
+
+        for i in range(0, 4):
+            byte_msg.append(self.mask_id[i])
+
+        byte_msg.append(CanAnalyzer.OPERATIONMODE[self.op_mode])
+        
+        byte_msg.append(0x01)
+
+        for i in range(0, 4):
+            byte_msg.append(0x00)
+
+        print "byte_str = " + str(byte_msg[2:])
+
+        crc = Crc8Darc.calc(byte_msg[2:])
+        crc_byte = struct.pack('B', crc)
+
+        print "crc_byte: " + str(crc_byte) + " length: " + str(len(crc_byte))
+
+        byte_msg.append(crc_byte)
+        self.ser.write(byte_msg)
+
+    def flush_buffer(self):
+        self.ser.flushInput()
+
+    def status_frame(self, timeout=None):
+        byte_msg = bytearray()
+        byte_msg.append(0xAA)     # Frame Start Byte 1
+        byte_msg.append(0x55)     # Frame Start Byte 2
+        byte_msg.append(0x04)     # Status Message ID
+        byte_msg.append(0x00)     # In response packet - Rx error count
+        byte_msg.append(0x00)     # In response packet - Tx error count
+
+        for i in range(0, 14):
+            byte_msg.append(0x00)
+
+        print "byte_str = " + str(byte_msg[2:])
+
+        crc = Crc8Darc.calc(byte_msg[2:])
+        crc_byte = struct.pack('B', crc)
+
+        print "crc_byte: " + str(crc_byte) + " length: " + str(len(crc_byte))
+
+        byte_msg.append(crc_byte)
+        self.ser.write(byte_msg)
+
+
+    def send(self, msg, timeout=None):
+        """
+        Send a message over the serial device.
+
+        :param can.Message msg:
+            Message to send.
+
+            .. note:: Flags like ``extended_id``, ``is_remote_frame`` and
+                      ``is_error_frame`` will be ignored.
+
+            .. note:: If the timestamp is a float value it will be converted
+                      to an integer.
+
+        :param timeout:
+            This parameter will be ignored. The timeout value of the channel is
+            used instead.
+
+        """
+        try:
+            timestamp = struct.pack('<I', int(msg.timestamp * 1000))
+        except struct.error:
+            raise ValueError('Timestamp is out of range')
+        try:
+            a_id = struct.pack('<I', msg.arbitration_id)
+        except struct.error:
+            raise ValueError('Arbitration Id is out of range')
+        byte_msg = bytearray()
+        byte_msg.append(0xAA)
+        for i in range(0, 4):
+            byte_msg.append(timestamp[i])
+        byte_msg.append(msg.dlc)
+        for i in range(0, 4):
+            byte_msg.append(a_id[i])
+        for i in range(0, msg.dlc):
+            byte_msg.append(msg.data[i])
+        byte_msg.append(0xBB)
+        self.ser.write(byte_msg)
```

Comment:
I don't see the format of the sent frame in the notes. It is a little strange that the transmitted frames use timestamps but the received do not.

Reply:
It could take a week or two for me to resolve this.  

I suspect you've picked up that I've started from serial_can.py, and that I not got to implementing the required send packet.  My current application is only required to read.

Reply:
Maybe not so long:
https://github.com/SeeedDocument/USB-CAN_Analyzer/blob/master/res/USB-CAN%20software%20and%20drive(v7.10)/Document/USB%20(Serial%20port)%20to%20CAN%20protocol%20defines.pdf